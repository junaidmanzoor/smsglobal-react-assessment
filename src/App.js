import React, { Component } from "react";
import { Route } from "react-router-dom";
import AppNavbar from "./AppNavbar";

import SendMessage from "./sendmessage";
import Settings from "./settings";
import Reports from "./reports";
import "./styles/App.css";

class App extends Component {
  render() {
    return (
      <div>
        <AppNavbar />
        <main role="main" className="container my-2">
          <Route path="/settings" component={Settings} />
          <Route path="/messages" component={SendMessage} />
          <Route path="/reports" component={Reports} />
        </main>
      </div>
    );
  }
}

export default App;
