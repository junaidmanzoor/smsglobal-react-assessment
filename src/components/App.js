import React, { Component } from 'react';
import {
  Button
} from "reactstrap"

import AppNavbar from './AppNavbar'
import NotesList from './NotesList';
import '../styles/App.css';
import NewNote from './NewNote';

class App extends Component {
  constructor(props){
    super(props)
    this.toggleNewNoteDlg = this.toggleNewNoteDlg.bind(this)
    this.state = {
      createNote: false
    }
  }

  componentDidMount(){
    this.props.actions.getNotes();
  }

  render() {
    const {notes, actions} = this.props
    return (
      <div>
        <AppNavbar />
        <main role="main" className="container mt-5">
          <Button
            outline
            color="primary"
            className="mb-2"
            onClick={this.toggleNewNoteDlg}>
            Create Note
          </Button>
          
          <NotesList
            notes={notes.collection}
            deleteNote={actions.deleteNote} />
          
          <NewNote
            isOpen={this.state.createNote}
            createNote={actions.createNote}
            onDismiss={this.toggleNewNoteDlg} />
        </main>
      </div>      
    );
  }

  toggleNewNoteDlg() {
    this.setState({
      createNote: !this.state.createNote
    })
}

}

export default App;
