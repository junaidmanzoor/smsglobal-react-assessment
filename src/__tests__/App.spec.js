import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from '../components/App';

test('renders without crashing', () => {
  let notes = {collection:[]}
  let actions = {getNotes: jest.fn(), deleteNote: jest.fn()}
  const output = shallow(<App notes={notes} actions={actions}/>)
  expect(shallowToJson(output)).toMatchSnapshot()
})

test('calls getNotes on mount', () => {
  let notes = {collection:[]}
  let actions = {getNotes: jest.fn(), deleteNote: jest.fn()}
  const output = shallow(<App notes={notes} actions={actions}/>)
  
  expect(actions.getNotes.mock.calls.length).toBe(1)
})

test('create note button opens new note dialog', () => {
  let notes = {collection:[]}
  let actions = {getNotes: jest.fn(), deleteNote: jest.fn()}
  const output = shallow(<App notes={notes} actions={actions}/>)
  
  expect(output.state().createNote).toBeFalsy
  output.find('Button').simulate('click')
  expect(output.state().createNote).toBeTruthy
})

test('New note dialog dismiss button closes the dialog', () => {
  let notes = {collection:[]}
  let actions = {getNotes: jest.fn(), deleteNote: jest.fn()}
  const output = shallow(<App notes={notes} actions={actions}/>)
  
  // set the createNote state to true
  output.find('Button').simulate('click')
  expect(output.state().createNote).toBeTruthy

  output.find('NewNote').simulate('dismiss')
  expect(output.state().createNote).toBeFalsy
})