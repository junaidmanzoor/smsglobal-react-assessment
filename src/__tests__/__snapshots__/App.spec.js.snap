// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <AppNavbar />
  <main
    className="container mt-5"
    role="main"
  >
    <Button
      className="mb-2"
      color="primary"
      onClick={[Function]}
      outline={true}
      tag="button"
    >
      Create Note
    </Button>
    <NotesList
      deleteNote={[MockFunction]}
      notes={Array []}
    />
    <NewNote
      isOpen={false}
      onDismiss={[Function]}
    />
  </main>
</div>
`;
